core/esmf-aspect-model-aas-generator/src/main/java/org/eclipse/esmf/aspectmodel/aas/AspectModelAasVisitor.java

```diff
@@ -568,28 +570,29 @@ private <T extends Collection> Environment visitCollectionProperty( final T coll
          return submodelBuilder.build();
       };
 
-      final Optional<JsonNode> rawValue = context.getRawPropertyValue();
-      return rawValue.map( node -> {
-         if ( node instanceof final ArrayNode arrayNode ) {
-            final SubmodelElementBuilder listBuilder = property -> {
-               final List<SubmodelElement> values = getValues( collection, property, context, arrayNode );
-               return new DefaultSubmodelElementList.Builder()
-                     .idShort( property.getName() )
-                     .displayName( LangStringMapper.NAME.map( property.getPreferredNames() ) )
-                     .description( LangStringMapper.TEXT.map( property.getDescriptions() ) )
-                     .value( values )
-                     .typeValueListElement( AasSubmodelElements.SUBMODEL_ELEMENT )
-                     .build();
-            };
-            createSubmodelElement( listBuilder, context );
-            return context.getEnvironment();
-         }
-         createSubmodelElement( builder, context );
-         return context.getEnvironment();
-      } ).orElseGet( () -> {
-         createSubmodelElement( builder, context );
-         return context.getEnvironment();
-      } );
+      final SubmodelElementBuilder listBuilder =
+            tryFindPropertyMapper( context.getProperty() )
+                  .flatMap( mapper -> collection.getDataType()
+                        .map( type -> (SubmodelElementBuilder) ( Property property ) -> mapper.mapToAasProperty( type, property,
```

Comment:
For nested expression code like this, please wrap after `->` to increase readability

Reply:
done.